### DISCUSSION TITLE
Recommendation on how to test a third party workload

### DISCUSSION THREAD

**Discussion by q-benwillis** [microsoft/Microsoft-Fabric-workload-development-sample#164] [General]
Hi all,

We're looking for ideas on how to properly test our third party workload.

Obviously we already have a full suite of unit tests, but are looking to improve our integration and end to end tests.

At the moment there's no REST API we can use to create a custom workload item in Fabric. Are there any suggestions on how we can run a backend integration test between our workload and Fabric? Without being able to create a new item in Fabric our requests to the permissions endpoints all fail as the item does not exist in Fabric.

Even then it seems the only way to get an access token from Fabric is using the javascript workload client. Is there an alternative that means we do not need to run our front end just to test the backend integration?

I'm keen to hear peoples thoughts!

Thanks

**Comment by IdanGoldfarb1989** [comment]
Unfortunately, no.
The only way to create your workload items is by using the JavaScript APIs provided in our SDK.

We are working on supporting public APIs, but there is no ETA yet.
